### PR TITLE
Include web book URL in add book form

### PR DESCRIPTION
**Pull request by jimchamp**
Adds field for a web book URL in the add book form.
Also adds a providers table to edition edit forms. New rows can be added to the table dynamically.

### Screenshot

Screenshot from 2022-10-13 17-13-53

no_providers

### CODE REVIEW

File: openlibrary/templates/books/edit/edition.html [hunk 1]
```diff
@@ -628,12 +628,58 @@
...
+        </div>
+    </fieldset>
+
 $if ctx.user and ctx.user.is_admin():
     <fieldset class="major adminOnly">
         <legend>$_("Admin Only")</legend>
         <div class="formElement">
             <div class="label">
-                <label for="edition-first_sentence">$_("Additional Metadata")</label>
+                <label for="additional_metadata">$_("Additional Metadata")</label>
```
Comment (jimchamp) [inline openlibrary/templates/books/edit/edition.html:682]: This is a bug-fix that is unrelated to this feature.  If uncorrected, clicking on the "Additional Metadata" label causes the "First sentence" textarea to gain focus.

**Comment by mekarpeles** [comment]
Awesome changes!

File: openlibrary/plugins/openlibrary/js/add_provider.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    </td>`
+}
+
+const accessTypeValues = [
+    {value: '', text: ''},
+    {value: 'read', text: 'Read'},
+    {value: 'listen', text: 'Listen'},
+    {value: 'buy', text: 'Buy'},
+    {value: 'borrow', text: 'Borrow'},
+    {value: 'preview', text: 'Preview'}
```
Comment (cdrini) [inline openlibrary/plugins/openlibrary/js/add_provider.js:45]: Oooh this schema isn't what the backend expects! I'll create a shim for now but we should make it consistent with what's in the google doc later.
